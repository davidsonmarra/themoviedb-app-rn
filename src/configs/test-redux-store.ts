import {configureStore} from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';

import reducers from '../../src/store/slices';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer: reducers,
  middleware: gDM => gDM().concat(sagaMiddleware),
});

export type IRootState = ReturnType<typeof reducers>;

export default store;
export {reducers, sagaMiddleware};
