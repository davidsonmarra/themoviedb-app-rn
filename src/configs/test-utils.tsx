import {render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import store, {IRootState, reducers, sagaMiddleware} from './test-redux-store';
import {configureStore, EnhancedStore} from '@reduxjs/toolkit';
import sagas from '../../src/store/sagas';

interface IAllThemeProviders {
  children: React.ReactElement;
  store?: EnhancedStore;
}

const AllTheProviders = ({children}: IAllThemeProviders) => (
  <Provider store={store}>{children}</Provider>
);

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export const renderWithCustomProviders = (
  component: React.ReactElement,
  preloadedState: IRootState,
) => {
  const reduxStore = configureStore({
    reducer: reducers,
    middleware: gDM => gDM().concat(sagaMiddleware),
    preloadedState: preloadedState,
  });

  sagaMiddleware.run(sagas);

  return {
    render: render(<Provider store={reduxStore}>{component}</Provider>),
    store: reduxStore,
  };
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
