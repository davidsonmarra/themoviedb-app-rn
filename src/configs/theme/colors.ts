export default {
  primary: '#01B4A4',
  primaryDark: '#032541',
  background: '#F1F1F1',
  title: '#121212',
  text: '#222222',
  buttonText: '#FFFFFF',
  border: '#595959',
  borderLight: '#CCCCCC',
  error: '#e3330b',
  success: '#00C851',
  warning: '#FFBB33',
  disabled: '#ABABAB',
  overlay: 'rgba(0,0,0,0.5)',
  overlayDark: 'rgba(0,0,0,0.75)',
  description: '#808080',
};
