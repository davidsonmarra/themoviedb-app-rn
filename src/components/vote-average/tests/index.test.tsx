import React from 'react';
import {render, screen} from '@testing-library/react-native';
import VoteAverage from '..';
import {Text} from '../..';
import getStyles from '../styles';
import textStyles from '../../text/styles';
import {colors} from '../../../configs/theme';

const mockBadAverage = 3;
const createComponent = (average: number = mockBadAverage) =>
  render(<VoteAverage voteAverage={average} />);

describe('Button', () => {
  it('renders correctly', () => {
    const tree = createComponent().toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders text correctly', () => {
    const instance = createComponent().root;
    expect(instance.findByType(Text)).toBeTruthy();
  });

  it('should render bad average', () => {
    createComponent();
    const averageContainer = screen.getByTestId('average-container');
    const averageText = screen.getByTestId('average-text');
    expect(averageContainer.props.style).toStrictEqual(
      getStyles(colors.error).container,
    );
    expect(averageText.props.style).toStrictEqual([
      textStyles.title,
      getStyles(colors.error).average,
    ]);
  });

  it('should render good average', () => {
    createComponent(8);
    const averageContainer = screen.getByTestId('average-container');
    const averageText = screen.getByTestId('average-text');
    expect(averageContainer.props.style).toStrictEqual(
      getStyles(colors.success).container,
    );
    expect(averageText.props.style).toStrictEqual([
      textStyles.title,
      getStyles(colors.success).average,
    ]);
  });

  it('should render warning average', () => {
    createComponent(5);
    const averageContainer = screen.getByTestId('average-container');
    const averageText = screen.getByTestId('average-text');
    expect(averageContainer.props.style).toStrictEqual(
      getStyles(colors.warning).container,
    );
    expect(averageText.props.style).toStrictEqual([
      textStyles.title,
      getStyles(colors.warning).average,
    ]);
  });
});
