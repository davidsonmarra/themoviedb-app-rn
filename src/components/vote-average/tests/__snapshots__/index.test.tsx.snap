// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "#e3330b",
      "borderRadius": 32,
      "borderWidth": 5,
      "flexDirection": "row",
      "height": 64,
      "justifyContent": "center",
      "width": 64,
    }
  }
  testID="average-container"
>
  <Text
    style={
      [
        {
          "color": "#121212",
          "fontSize": 24,
          "fontWeight": "bold",
        },
        {
          "color": "#e3330b",
          "fontSize": 28,
        },
      ]
    }
    testID="average-text"
  >
    3.0
  </Text>
</View>
`;
